<launch>
  <!-- Common parameters -->
  <arg name="robot1_name" default="robot1"/>
  
  <!-- Robot positions -->
  <arg name="robot1_x" default="0.0"/>
  <arg name="robot1_y" default="0.0"/>
  <arg name="robot1_z" default="0.04"/>

  <!-- Start Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find grad)/worlds/real_warehouse_2.world"/>
    <arg name="gui" value="true"/>
  </include>

  <!-- Robot 1 -->
  <group ns="$(arg robot1_name)">
    <!-- Load URDF into PARAMETER SERVER -->
    <param name="robot_description" 
           command="$(find xacro)/xacro '$(find grad)/urdf/robot.urdf.xacro' 
           robot_namespace:=$(arg robot1_name)"/>

    <!-- Spawn robot in Gazebo -->
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -model $(arg robot1_name) 
                -param /$(arg robot1_name)/robot_description
                -x $(arg robot1_x) -y $(arg robot1_y) -z $(arg robot1_z)"/>

    <!-- Robot State Publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
      <remap from="robot_description" to="robot_description"/>
      <param name="tf_prefix" value="$(arg robot1_name)"/>
    </node>

    <!-- Joint State Publisher -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="tf_prefix" value="$(arg robot1_name)"/>
    </node> 
  </group>


  
  <!-- Cartographer Node -->
 
  <!-- <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
        type="cartographer_occupancy_grid_node" args="-resolution 0.05">
        <remap from="odom" to="robot1/odom" />
  </node>

  <arg name="laser_scan_topic" default="/robot1/robot1/scan"/>

    
  <node name="cartographer_node" pkg="cartographer_ros"
      type="cartographer_node" args="
          -configuration_directory $(find grad)/config
          -configuration_basename my_robot.lua"
      output="screen">
      
      <remap from="odom" to="robot1/odom" />
      <remap from="/scan" to="$(arg laser_scan_topic)"/>
      <remap from="base_link" to="robot1/base_footprint" />

  </node>
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cartographer_ros)/configuration_files/demo_2d.rviz" /> -->


<!-- 
  <node name="slam_gmapping" pkg="gmapping" type="slam_gmapping" output="screen">
    <remap from="scan" to="/robot1/robot1/scan" /> 
    <param name="base_frame" value="robot1/base_footprint" />
    <param name="odom_frame" value="robot1/odom" />
    <param name="map_update_interval" value="5.0" />
    <param name="maxUrange" value="6.0" />
    <param name="sigma" value="0.05" />
    <param name="kernelSize" value="1" />
    <param name="lstep" value="0.05" />
    <param name="astep" value="0.05" />
    <param name="iterations" value="5" />
    <param name="lsigma" value="0.075" />
    <param name="ogain" value="3.0" />
    <param name="lskip" value="0" />
    <param name="minimumScore" value="200" />
    <param name="srr" value="0.01" />
    <param name="srt" value="0.02" />
    <param name="str" value="0.01" />
    <param name="stt" value="0.02" />
    <param name="linearUpdate" value="0.5" />
    <param name="angularUpdate" value="0.436" />
    <param name="temporalUpdate" value="-1.0" />
    <param name="resampleThreshold" value="0.5" />
    <param name="particles" value="80" />
    <param name="xmin" value="-10.0" />
    <param name="ymin" value="-10.0" />
    <param name="xmax" value="10.0" />
    <param name="ymax" value="10.0" />
    <param name="delta" value="0.05" />
    <param name="llsamplerange" value="0.01" />
    <param name="llsamplestep" value="0.01" />
    <param name="lasamplerange" value="0.005" />
    <param name="lasamplestep" value="0.005" />
  </node> 

  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find grad)/rviz/gmapping.rviz" output="screen" /> -->


</launch>